src/main/java/com/github/kongchen/swagger/docgen/mavenplugin/ApiSource.java

```diff
@@ -134,28 +134,32 @@
 
     @Parameter
     private List<String> modelConverters;
+    
+    @Parameter
+    private boolean skipInheritingClasses = false;
 
     public Set<Class<?>> getValidClasses(Class<? extends Annotation> clazz) {
         Set<Class<?>> classes = new LinkedHashSet<Class<?>>();
+        
+        List<String> prefixes = new ArrayList<String>();
         if (getLocations() == null) {
-            Set<Class<?>> c = new Reflections("").getTypesAnnotatedWith(clazz, true);
-            classes.addAll(c);
-
-            Set<Class<?>> inherited = new Reflections("").getTypesAnnotatedWith(clazz);
-            classes.addAll(inherited);
+            prefixes.add("");
         } else {
-            for (String location : locations) {
-                Set<Class<?>> c = new Reflections(location).getTypesAnnotatedWith(clazz, true);
-                classes.addAll(c);
+            prefixes.addAll(getLocations());
+        }
+        
+        for (String location : prefixes) {
+            Set<Class<?>> c = new Reflections(location).getTypesAnnotatedWith(clazz, true);
```

Comment:
If i am not mistaken, scanning with an empty string will search, basicly, everywhere for types?
Is this also the current behaviour, when no location is defined within the ```<locations></locations>``` element?

Reply:
Yes, behavior is the same as before.